lib/Yield/Assembly/PTML.cpp

```diff
@@ -33,6 +33,9 @@ static constexpr auto ImmediateValue = "asm.immediate-value";
 static constexpr auto MemoryOperand = "asm.memory-operand";
 static constexpr auto Register = "asm.register";
 
+static constexpr auto FunctionLink = "call-graph.function-link";
+static constexpr auto ShallowFunctionLink = "call-graph.shallow-function-link";
```

Comment:
* Why do you have "shallow" and "non-shallow" function links? Links are just links, they are always shallow. I think there's some confusion about definition and reference here.
* A token type should not be coupled to how you plan to make it interactive. The token should just be `function-identifier`.
* Why is this in `Assembly/*` if it's for the call graph only?

Reply:
1-2. Yeah, I'm not satisfied with the naming either. But we do in fact need a way to differentiate between the "real" nodes and the nodes we're adding as a duplicates without their subtrees attached. This is probably not the right way, but it feels like the easiest option for the client to easily make them visually distinct.

3. Yeah, sorry. I'll move them closer to the users.

Reply:
`1-2` so you want to have them visually different in the call graph uh?
What about `call-graph.function-identifier` and `call-graph.function-reference` or `call-graph.shallow-function-identifier`?

Reply:
1-2. I feel like that naming scheme is even more confusing. What about `call-graph.function-reference` for the proper reference (after all "reference" is no different from a "link" but does sound better, imo) and then add something like "severed" or "excluded" or "displaced" or something like that to the name (I feel like it's a better explanation of what has happened to it over the "shallow" thing), don't you?

Reply:
Actually, yes! That sounds great, they don't have to be functions per se. But also, do we need to differentiate between definition and reference in the token? We have special fields for that after all. What about `call-graph.node-label` for both "real" definitions and references and `call-graph.shallow-node-label` for shallow references?

Reply:
Works!

Reply:
Done: https://github.com/revng/revng/pull/191/commits/06d99706bc7b97797cc8268e271850a8c0ec706f.